File: map.js
// import React, { useState, useEffect } from 'react';
// import MapView, { Marker } from 'react-native-maps';
// import { Directions } from 'react-native-directions';
// import * as Location from 'expo-location';
// import { View, Text } from 'react-native';

// const DirectionsScreen = () => {
//   const [currentLocation, setCurrentLocation] = useState(null);
//   const [destination, setDestination] = useState({
//     latitude: -1.272726,
//     longitude: 36.71989,
//     title: 'Kiambu',
//   });

//   useEffect(() => {
//     (async () => {
//       const location = await Location.getCurrentPositionAsync({});
//       setCurrentLocation(location);
//     })();
//   }, []);

//   return (
//     <View style={{ flex: 1 }}>
//       <MapView
//         style={{ flex: 1 }}
//         region={{
//           latitude: currentLocation?.coords.latitude ?? 0,
//           longitude: currentLocation?.coords.longitude ?? 0,
//           latitudeDelta: 0.02,
//           longitudeDelta: 0.02,
//         }}
//         showsUserLocation={true}
//       >
//         <Marker
//           coordinate={{
//             latitude: currentLocation?.coords.latitude ?? 0,
//             longitude: currentLocation?.coords.longitude ?? 0,
//           }}
//           title="Current Location"
//         />
//         <Marker coordinate={destination} title="Kiambu" />
//         <Directions
//           origin={currentLocation}
//           destination={destination}
//           apikey={'s'} // Replace this with your Google Maps API key
//           strokeWidth={3}
//           strokeColor="hotpink"
//         />
//       </MapView>
//     </View>
//   );
// };

// export default DirectionsScreen;

File: map.py
# import googlemaps
# from gmplot import gmplot

# # Replace with your Google Maps API Key
# api_key = 'AIzaSyAE3VC57Az0LKqEr4vTnxxxobz7yudIqTEy'

# # Initialize Google Maps client
# gmaps = googlemaps.Client(key=api_key)

# # Get the coordinates (latitude and longitude) of the two locations
# location1 = gmaps.geocode("Location 1")[0]['geometry']['location']
# location2 = gmaps.geocode("Location 2")[0]['geometry']['location']

# # Calculate the distance between the two locations
# distance = gmaps.distance_matrix(location1, location2)['rows'][0]['elements'][0]['distance']['text']

# # Create a Google Maps plot
# gmap = gmplot.GoogleMapPlotter(location1['lat'], location1['lng'], zoom=10)

# # Add markers for both locations
# gmap.marker(location1['lat'], location1['lng'], 'cornflowerblue', title='Location 1')
# gmap.marker(location2['lat'], location2['lng'], 'red', title='Location 2')

# # Create an HTML file to display the map
# gmap.draw("map.html")

# print(f"Distance between Location 1 and Location 2: {distance}")
